packages/cli/src/commands/deviceprofiles/create.ts

```diff
@@ -42,3 +89,146 @@ export function cleanupRequest(deviceProfileRequest: Partial<DeviceProfile & { r
+			}
+
+			let profile = await this.client.deviceProfiles.create(cleanupRequest(data))
+
+			if (!data.metadata?.vid) {
+				// Generate the default config
+				const deviceConfigData = await generateDefaultConfig(this.client, profile.id, profile)
+
+				// Create the config using the default
+				const deviceConfig = await this.client.presentation.create(deviceConfigData)
+
+				// Update the profile to use the vid from the config
+				const profileId = profile.id
+				cleanupRequest(profile)
+				delete profile.name
+				if (!profile.metadata) {
+					profile.metadata = {}
+				}
+				profile.metadata.vid = deviceConfig.vid
+				profile.metadata.mnmn = deviceConfig.mnmn
+				profile = await this.client.deviceProfiles.update(profileId, profile)
+			}
+
+			return profile
+		})
+	}
+
+	protected async promptAndAddCapability(deviceProfile: DeviceProfileRequest, componentId: string, message = 'Capability ID'): Promise<void> {
+		const capabilityId: string = (await inquirer.prompt({
+			type: 'input',
+			name: 'capabilityId',
+			message: message + ': ',
+			validate: (input) => {
+				return input.length > 0 || 'Invalid capability name'
+			},
+		})).capabilityId
+
+		if (capabilityId) {
+			const component = deviceProfile.components?.find(it => it.id === componentId)
+			if (component) {
+				component.capabilities?.push({id: capabilityId, version: 1})
+			} else {
+				throw new Error(`Component ${componentId} not defined in profile`)
+			}
+		}
+	}
+
+	protected async promptAndAddComponent(deviceProfile: DeviceProfileRequest, previousComponentId: string): Promise<string> {
+		const components = deviceProfile.components || []
+		let componentId: string = (await inquirer.prompt({
+			type: 'input',
+			name: 'componentId',
+			message: 'ComponentId ID: ',
+			validate: (input) => {
+				return (new RegExp(/^[0-9a-zA-Z]{1,100}$/).test(input) && !components.find(it => it.id === input)) || 'Invalid component name'
+			},
+		})).componentId
+
+		if (componentId) {
+			components.push({id: componentId, capabilities: []})
+		} else {
+			componentId = previousComponentId
+		}
+		return componentId
+	}
+
+	protected async getInputFromUser(): Promise<DeviceProfileRequest> {
+		const name = (await inquirer.prompt({
+			type: 'input',
+			name: 'deviceProfileName',
+			message: 'Device Profile Name:',
+			validate: (input: string) => {
+				return new RegExp(/^(?!\s)[-_!.~'() *0-9a-zA-Z]{1,100}(?<!\s)$/).test(input) ||  'Invalid device profile name'
+			},
+		})).deviceProfileName
+
+		const deviceProfile: DeviceProfileRequest = {
+			name,
+			components: [
+				{
+					id: 'main',
+					capabilities: [],
+				},
+			],
+		}
+
+		await this.promptAndAddCapability(deviceProfile, 'main', 'Primary capability ID')
+
+		const enum Action {
+			ADD_CAPABILITY = 'Add another capability to this component',
+			ADD_COMPONENT = 'Add another component',
+			FINISH = 'Finish & Create',
+		}
```

Comment:
I like this idea but maybe it could live outside of this function in a util collection? Unless other inquirer commands would not benefit from them, but it seems like they are generic enough.

Reply:
I'm conflicted about breaking these out. Do we wind up with all the text for all commands in one file?  It loses the context. Right now the only other place we do this is in creating capabilities, and the only text in common is "Finish & Create". I'd like to leave this as it is for this PR, and discuss the pros and cons of pulling out user text from the command classes

Reply:
I agree that if they really aren't that reusable from other commands, we should leave them be.